src/dstack/_internal/cli/commands/offer.py

```
@@ -0,0 +1,199 @@
+from dstack.api.utils import load_profile
+
+
+class OfferConfigurator(BaseRunConfigurator):
+    # TODO: The command currently uses `BaseRunConfigurator` to register arguments.
+    #   This includes --env, --retry-policy, and other arguments that are unnecessary for this command.
+    #   Eventually, we should introduce a base `OfferConfigurator` that doesn't include those arguments—
+    #   `BaseRunConfigurator` will inherit from `OfferConfigurator`.
+    #
+    #   Additionally, it should have its own type: `ApplyConfigurationType.OFFER`.
+    TYPE = ApplyConfigurationType.TASK
+
+    @classmethod
+    def register_args(
+        cls,
+        parser: argparse.ArgumentParser,
+    ):
+        super().register_args(parser, default_max_offers=50)
+
+
+# TODO: Support aggregated offers
+# TODO: Add tests
+class OfferCommand(APIBaseCommand):
+    NAME = "offer"
+    DESCRIPTION = "List offers"
+
+    def _register(self):
+        super()._register()
+        self._parser.add_argument(
+            "--format",
+            choices=["plain", "json"],
+            default="plain",
+            help="Output format (default: plain)",
+        )
+        self._parser.add_argument(
+            "--json",
+            action="store_const",
+            const="json",
+            dest="format",
+            help="Output in JSON format (equivalent to --format json)",
+        )
+        OfferConfigurator.register_args(self._parser)
+
+    def _command(self, args: argparse.Namespace):
+        super()._command(args)
+        conf = TaskConfiguration(commands=[":"])
+
+        configurator = OfferConfigurator(api_client=self.api)
+        configurator.apply_args(conf, args, [])
+        profile = load_profile(Path.cwd(), profile_name=args.profile)
+
+        if args.format == "json":
+            run_spec = RunSpec(
+                configuration=conf,
+                ssh_key_pub=Path(self.api.ssh_identity_file + ".pub").read_text().strip(),
```

(nit) I think you can use a dummy value to avoid reading from disk and avoid unnecessary errors if the key is misconfigured.

Same on line 84